python2verilog/backend/verilog/codegen.py

```diff
@@ -494,26 +490,34 @@ def __init__(self, root: ir.Node, context: ir.Context):
         self.visited: set[str] = set()
         self.context = context
         self.case = ver.Case(expression=context.state_var, case_items=[])
+        self.root = root
 
         # Member Funcs
         instance = itertools.count()
         self.next_unique = lambda: next(instance)
 
-        # Work
-        logging.debug(f"{self.__class__.__name__} {root.unique_id} {root}")
-        self.case.case_items.append(self.new_caseitem(root))
-        has_done = False
+    def get_case(self) -> ver.Case:
+        """
+        Gets case statement/block
+        """
+        # Start recursion and create FSM
+        self.case.case_items.append(self.new_caseitem(self.root))
+
+        # Reverse states for readability (states are built backwards)
+        self.case.case_items = list(reversed(self.case.case_items))
+
+        # Add done state if it doesn't exist
         for case in self.case.case_items:
-            if case.condition == context.done_state:
-                has_done = True
-        if not has_done:
+            if case.condition == self.context.done_state:
+                break
+        else:  # no break
```

Comment:
Maybe a bit clearer: 

```python
done_state = find(lambda case : case.condition == self.context.done_state, self.case.case_items)

if not done_state:
    self.case.case_items.append(
```

Reply:
There's no find function in Python, but I'm assuming you meant something like
`done_state = (case for case in self.case.case_items if case.condition == self.context.done_state)`.
However, I think the current code is more readable.

Reply:
I dislike the need for the `else: # no break` statement. I think something with a flag will be more readable (even if it means reverting back to the old code).